docs/src/book.md

```diff
@@ -0,0 +1,178 @@
+source concurrently with the present text.
+
+To start we load the generic character table for $\mathrm{SL}_3(q)$ with
+$q\not\equiv 1\pmod 3$.
+To learn about the origin of the table, we could enter `print(info(T))`.
+```jldoctest book
+julia> T = genchartab("SL3.n1")
+Generic character table SL3.n1
+  of order q^8 - q^6 - q^5 + q^3
+  with 8 irreducible character types
+  with 8 class types
+  with parameters (a, b, m, n)
+```
+
+As we can see this table has four parameters in addition to $q$.
+The entries of the table are “generalized cyclotomics”, that is,
+linear combinations over $\mathbb{Q}(q)$ of symbolic “roots of unity”
+depending on the parameters listed above.
+```jldoctest book
+julia> T[4,4]
+(q + 1)*exp(2π𝑖((a*n)//(q - 1))) + exp(2π𝑖((-2*a*n)//(q - 1)))
+```
+
+Denoting row 4 of the table by $\chi_4$, we note that is not a single
+character, but a *character type*, describing a whole family of
+characters.
+```jldoctest book
+julia> χ₄ = T[4]
+Generic character of SL3.n1
+  with parameters
+    n ∈ {1,…, q - 1} except n ∈ (q - 1)ℤ
+  of degree q^2 + q + 1
+  with values
+    q^2 + q + 1
+    q + 1
+    1
+    (q + 1)*exp(2π𝑖((a*n)//(q - 1))) + exp(2π𝑖((-2*a*n)//(q - 1)))
+    exp(2π𝑖((a*n)//(q - 1))) + exp(2π𝑖((-2*a*n)//(q - 1)))
+    exp(2π𝑖((a*n)//(q - 1))) + exp(2π𝑖((b*n)//(q - 1))) + exp(2π𝑖((-a*n - b*n)//(q - 1)))
+    exp(2π𝑖((a*n)//(q - 1)))
+    0
+```
+
+With these generic characters, we can compute norms, scalar products, and more.
+For this demonstration we tensor the second character type $\chi_2$ with itself.
+```jldoctest book
+julia> h = T[2] * T[2]
+Generic character of SL3.n1
+  of degree q^4 + 2*q^3 + q^2
+  with values
+    q^4 + 2*q^3 + q^2
+    q^2
+    0
+    q^2 + 2*q + 1
+    1
+    4
+    0
+    1
+```
+
+We may now attempt to decompose this character type `h` by computing its scalar product with the irreducible character types.
+This returns a “generic” scalar product, plus a (possibly empty)
+list of parameter exceptions for which the general result may not hold.
+For example:
+```jldoctest book
+julia> scalar_product(T[4], h)
+0
+With exceptions:
+  2*n1 ∈ (q - 1)ℤ
+```
+This scalar product is $0$
+except when $q-1$ divides $2n_1$, where $n_1$ indicates the value of the parameter $n$ for the first factor in the scalar product (i.e. the character type $\chi_4$).
+For a “generic decomposition” we need to compute all the scalar products and
+exceptions.
+```jldoctest book
+julia> for i in 1:8 println("<$i, h> = ", scalar_product(T[i], h)) end
+<1, h> = 1
+<2, h> = 2
+<3, h> = 2
+<4, h> = 0
+With exceptions:
+  2*n1 ∈ (q - 1)ℤ
+<5, h> = 0
+With exceptions:
+  2*n1 ∈ (q - 1)ℤ
+<6, h> = 0
+With exceptions:
+  2*m1 - n1 ∈ (q - 1)ℤ
+  m1 - 2*n1 ∈ (q - 1)ℤ
+  m1 + n1 ∈ (q - 1)ℤ
+  m1 ∈ (q - 1)ℤ
+  m1 - n1 ∈ (q - 1)ℤ
+  n1 ∈ (q - 1)ℤ
+<7, h> = 0
+With exceptions:
+  n1 ∈ (q - 1)ℤ
+<8, h> = 0
+With exceptions:
+  q*n1 ∈ (q^2 + q + 1)ℤ
+  n1 ∈ (q^2 + q + 1)ℤ
+  q*n1 + n1 ∈ (q^2 + q + 1)ℤ
```

Comment:
Off-topic, but: I think this can be simplified to just
```suggestion
  n1 ∈ (q^2 + q + 1)ℤ
```
because $q$ resp. $q+1$ both are coprime with $q^2 + q + 1$ for any value of $q$. And *that* could even be determined using polynomial gcd, i.e. I think there is a chance of automating this kind of simplification: just factor the left side; then kill any factors which are provably coprime to the "modulus". 

Actually it's not simply computing a `gcd`, it is a bit more: if we start with e.g. `gcd(q, q^2+q+2)`, what I'd really want is to simplify this to `gcd(q,2)`. At that point we arrive at a case distinction: the gcd is 1 if $q$ is odd, and it is $2$ if $q$ is even.

Reply:
> because q resp. q + 1 both are coprime with q 2 + q + 1 for any value of q . And that could even be determined using polynomial gcd, i.e. I think there is a chance of automating this kind of simplification: just factor the left side; then kill any factors which are provably coprime to the "modulus".

Yes, this should be easy to implement.

> Actually it's not simply computing a gcd, it is a bit more: if we start with e.g. gcd(q, q^2+q+2), what I'd really want is to simplify this to gcd(q,2). At that point we arrive at a case distinction: the gcd is 1 if q is odd, and it is 2 if q is even.

So, you are simplifying $q^2+q+2 \in q\mathbb{Z}$? As exceptions only occur in `nesum` and are always part of an exponent of a generic cyclotomic, these kind of exceptions will already be simplified. Internally the exception is $\frac{q^2+q+2}{q}=q+1+\frac{2}{q}$. And the exponents are only unique up to integers so this would be simplified to $\frac{2}{q}$.

Reply:
I was simplifying a hypothetical `q\in (q^2+q+2)\mathbb{Z}` -- simply looked at the "problem" here and thought how one could handle it programmatically; and then thinking about ways to tweak the input to break what I just came up with -- hence arrive at this artificial example. I mainly wanted to explain why we shouldn't use `gcd(q,2)=1` (which is what Oscar gives us).

Anyway, this is for the future.